controllers/servers/host_definer/watcher/watcher_helper.py

```diff
@@ -0,0 +1,263 @@
+        return False
+
+    def _get_host_definition_phase(self, host_definition):
+        if host_definition.status:
+            return host_definition.status.phase
+        return ''
+
+    def verify_host_defined_on_storage_and_on_cluster(self, host_definition):
+        response = self._ensure_definition_state(host_definition, self.storage_host_servicer.define_host)
+        if response.error_message:
+            return response
+        self._verify_host_definition_in_phase(host_definition, settings.READY_PHASE)
+        return response
+
+    def _verify_host_definition_in_phase(self, host_definition, phase):
+        host_definition_manifest = self.get_host_definition_manifest(host_definition)
+        logger.info('Verifying hostDefinition {} is in {} phase'.format(host_definition.name, phase))
+        host_definition_instance, _ = self._get_host_definition(host_definition.name)
+        if host_definition_instance:
+            self.patch_host_definition(host_definition_manifest)
+        else:
+            logger.info('Creating host Definition: {}'.format(host_definition.name))
+            self._create_host_definition(host_definition_manifest)
+        self.set_host_definition_status(host_definition.name, phase)
+
+    def get_host_definition_name(self, management_address, node_name):
+        return '{0}.{1}'.format(management_address, node_name).replace('_', '.')
+
+    def get_host_definition_manifest(self, host_definition):
+        return {
+            'apiVersion': settings.CSI_IBM_API_VERSION,
+            'kind': settings.HOST_DEFINITION_KIND,
+            'metadata': {
+                'name': host_definition.name,
+            },
+            'spec': {
+                'hostDefinition': {
+                    settings.MANAGEMENT_ADDRESS_FIELD: host_definition.management_address,
+                    settings.NODE_NAME_FIELD: host_definition.node_name,
+                    settings.NODE_ID_FIELD: host_definition.node_id,
+                    settings.SECRET_NAME_FIELD: host_definition.secret_name,
+                    settings.SECRET_NAMESPACE_FIELD: host_definition.secret_namespace,
+                },
+            },
+        }
+
+    def _get_prefix(self):
+        return os.getenv('PREFIX')
+
+    def get_connectivity(self):
+        return os.getenv('CONNECTIVITY')
+
+    def undefine_host_and_host_definition_with_events(self, host_definition):
+        node_name = host_definition.node_name
+        logger.info('Verifying that host {} is undefined from storage {}'.format(node_name,
+                                                                                 host_definition.management_address))
+        response = self.undefine_host_and_host_definition(host_definition)
+        if response.error_message:
+            self.set_host_definition_status(host_definition.name, settings.PENDING_DELETION_PHASE)
+            self._create_event_to_host_definition(host_definition.name, response.error_message)
```

Comment:
please change similarly to [here](https://github.com/IBM/ibm-block-csi-driver/pull/514#discussion_r908741892)

Reply:
done